lib/SIL/Verifier/SILVerifierCAPI.h

```diff
@@ -0,0 +1,69 @@
+//===--- SILVerifierCAPI.h ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// C API for calling the SILVerifier from a dynamically loaded dylib.
+///
+//===----------------------------------------------------------------------===//
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef SILVERIFIERCAPI_WEAK
+#error "SILVERIFIERCAPI_WEAK defined?!"
+#endif
+
+#ifdef __APPLE__
+#define SILVERIFIERCAPI_WEAK __attribute__((weak_import))
+#elif LINUX
+#define SILVERIFIERCAPI_WEAK __attribute__((weak))
+#else
+#error "Unhandle platform?!"
```

Comment:
This 100% does not work on all platforms - Windows does not have the concept of weak linking.  Why not do something like a series of `dlsym` (or the moral equivalent `GetProcAddr`) to get a table of function pointers?